### DISCUSSION TITLE
Get calendar events for specific timeframe

### DISCUSSION THREAD

**Discussion by kurgompro** [derekantrican/GAS-ICS-Sync#413] [Q&A]
Dear Community,

I want to fetch calendar events for a given timeframe. In my case, I would like to get events FROM the past 30 days TO the future 365 days. For the past 30 days, I tried to set 'onlyFutureEvents' to 'true' and change 'startUpdateTime' to 'new ICAL.Time.fromJSDate(new Date()) - 30', without luck.

Any ideas on how to achieve this in a stable way?

Thanks,
kurgompro

**Comment by kurgompro** [comment]
I found a way to also fetch the past 5 days when 'onlyFutureEvents' is 'true'. At line 158, I adjusted as follows:

```
if (onlyFutureEvents)
startUpdateTime = new ICAL.Time.fromJSDate(new Date(new Date().getTime() - ((5)*(24*60*60*1000))));
```

where (5) is the number of days and (24x60x60x1000) is the calculation for 1 day.

What I am still looking for is the also limit the future events fetching to 365 days, but also need a way to capture this more user-friendly by using a variable. Any help would be appreciated. :-))

**Comment by jonas0b1011001** [comment]
Either use [this branch](https://github.com/derekantrican/GAS-ICS-Sync/tree/event-filtering) with filters.gs:

```
var filters = [
            {
              parameter: "dtend",
              type: "include", //include/exclude
              comparison: ">", // <,>, =
              offset: -30
            },
            {
              parameter: "dtstart",
              type: "exclude", //include/exclude
              comparison: ">", // <,>, =
              offset: 365
            } 
          ];
```

or have a look at #221

**Reply by kurgompro** [reply]
Hi @jonas0b1011001 ,

That is very helpful. I have used the solution outlined in https://github.com/derekantrican/GAS-ICS-Sync/issues/129 and tweaked with a dynamic date as outlined in https://github.com/derekantrican/GAS-ICS-Sync/issues/221 - **Works perfectly - Thanks a lot**!

I was thinking of adding these customizations in a separate 'User.gs' and call it in 'Code.gs'. I will check if that could work, unless you say right away if that will cause issues.
